From cabb48e339a1f646b4a820d41eaef5c34e678708 Mon Sep 17 00:00:00 2001
From: Mohammed Naser <mnaser@vexxhost.com>
Date: Fri, 21 Feb 2025 15:20:18 -0500
Subject: [PATCH] Add MTU for all non-external ports

In the previous fix, it had brought the MTU lookup within the
if branch if the vnic_type was VNIC_REMOTELY_MANAGED which
meant that it didn't get added for anything other than that
typer of interface.

Since the MTU is used further later, we want to capture it
because it goes into the OvnPortInfo which will go into the
external_ids which will be captured by the metadata agent
to fix the correct MTU.

Closes-Bug: #2053274
Related-Change-Id: I7ff300e9634e5e3fc68d70540392109fd8b9babc
Change-Id: Ib4ee7e0e8b13141739f5a8600333be856afd4a4e
---

diff --git a/neutron/plugins/ml2/drivers/ovn/mech_driver/ovsdb/ovn_client.py b/neutron/plugins/ml2/drivers/ovn/mech_driver/ovsdb/ovn_client.py
index aa1c989..f18b558 100644
--- a/neutron/plugins/ml2/drivers/ovn/mech_driver/ovsdb/ovn_client.py
+++ b/neutron/plugins/ml2/drivers/ovn/mech_driver/ovsdb/ovn_client.py
@@ -473,11 +473,11 @@
         # HA Chassis Group will bind the port to the highest
         # priority Chassis
         if port_type != ovn_const.LSP_TYPE_EXTERNAL:
+            port_net = self._plugin.get_network(context, port['network_id'])
+            mtu = str(port_net['mtu'])
+
             if (bp_info.vnic_type == portbindings.VNIC_REMOTE_MANAGED and
                     ovn_const.VIF_DETAILS_PF_MAC_ADDRESS in bp_info.bp_param):
-                port_net = self._plugin.get_network(
-                    context, port['network_id'])
-                mtu = str(port_net['mtu'])
                 options.update({
                     ovn_const.LSP_OPTIONS_VIF_PLUG_TYPE_KEY: 'representor',
                     ovn_const.LSP_OPTIONS_VIF_PLUG_MTU_REQUEST_KEY: mtu,
